### DISCUSSION TITLE
I want to discuss my proposal w.r.t the Structured Transparency Framework with a mentor

### DISCUSSION THREAD

**Discussion by rahulsharma-rs** [OpenMined/courses#253] [Project: Privacy Product Specification] [answered]
Hi Mentors,
I finished my Private AI beginners course but failed in the project part in my first attempt.
I have some specific doubts related to my proposal if any of the mentors take some time and guide me through I will be highly obliged.
I want to discuss my proposal w.r.t the Structured Transparency Framework.
Hope to hear one of the mentors soon.
Thanks in advance.
Rahul. (edited)

**Comment by ChetanMadan** [comment]
Hey, @rahulsharma-rs!
Congratulations on finishing the course. You can go ahead and ask any doubts you have in this thread!

**Comment by carrycooldude** [comment]
> Hi Mentors,
> I finished my Private AI beginners course but failed in the project part in my first attempt.
> I have some specific doubts related to my proposal if any of the mentors take some time and guide me through I will be highly obliged.
> I want to discuss my proposal w.r.t the Structured Transparency Framework.
> Hope to hear one of the mentors soon.
> Thanks in advance.
> Rahul. (edited)

I have same doubts

**Reply by ChetanMadan** [reply]
Same?
Anyway, if you have some specific questions, you can post them here, and I and the other mentors will do our best to guide you through!

**Reply by rahulsharma-rs** [reply]
The first doubt is related to Federated Learning since there is two-way communication between the ML engineer and the Data provider, therefore do we have two input privacy scenarios, one at the engineer's side and the other at the Data provider side? Similarly, do we have two output privacy scenarios?

**Reply by ChetanMadan** [reply]
In the case of Federated Learning, generally, the information flows are in terms of model parameters (weights and biases), being sent to and from a local and a global model and  "sensitive" data is not sent across at all. So there is no need for two input and output privacy scenarios.

**Reply by rahulsharma-rs** [reply]
@ChetanMadan I think, I am doing something fundamentally wrong. Is it possible to get on a call? I can explain my confusions in depth

**Reply by leriomaggio** [reply]
@rahulsharma-rs RE: "do we have two input privacy scenarios"
Considerations about the Structured transparency frameworks have to be carried out _per-single_ information flow, _iow_:
"given a specific information flow you have in mind // you're designing", define the characteristics and constraints of  the 5 components of the framework.

HTH.

RE: "is it possible to get on a call?"
I would recommend keeping conversations open here, and publicly available to everyone so that others that share some doubts or concerns can benefit from mentors' support as well :) 

So again, as also suggested by @ChetanMadan, please feel free to post any question or doubt you may have here - we are happy to help!

**Comment by manik-500** [comment, marked as answer]
hi @rahulsharma-rs  
In the case of federated learning your ml model will send to the device where the users data lives . As the user can copy your model , so you have input privacy issues here .You can encrypt your models by Homomorphic Encryption or secure multiparty computation to prevent this .
As the data is not moving from their respective device  so nobody can copy users data . So in this case input privacy is not needed for the users data .
So in federated learning you have one input privacy scenarios not two .

**Reply by rahulsharma-rs** [reply]
Hi @manik-500
Thank you for the clarification. 
What about output privacy? Is it two in this scenario? and what tools we require to preserve this output privacy?
Thanks in advance :)
Rahul.

**Reply by manik-500** [reply]
Hi @rahulsharma-rs 
The ML engineer can request the output of the models for debugging purposes . And from the output of the ml models they (ML engineer) can use reverse engineering to infer input data or other attributes of input data . So you have output privacy issues here . you can use differential privacy (add random noise on input data according to privacy budget Epsilon ) to prevent unintended information leakage for addressing output privacy .

For the second case as the ml model will be encrypted and there’s nothing about anything from where users can use reverse engineering to infer models parameters (weight and bias) .So in this case i think output privacy is not needed .

BTW nice question . If you have further questions , you can ask without hesitation . We are all here to help you .
Thanks .
